ui/embed.go

```diff
@@ -26,35 +26,44 @@ func ServeFile(w http.ResponseWriter, r *http.Request, filePath string) error {
 }
 
 // ServePage serves the index.html page.
-func ServePage(w http.ResponseWriter, r *http.Request, page string, data map[string]interface{}) error {
+func ServePage(w http.ResponseWriter, r *http.Request, page, title string, data map[string]interface{}) error {
 	if data == nil {
-		data = make(map[string]interface{})
+		data = make(map[string]any)
 	}
 	data["csrfToken"] = csrf.Token(r)
 	data["page"] = page
 
-	jsonData, err := json.Marshal(data)
+	bs, err := renderTemplate("dist/index.gohtml", map[string]any{
+		"Title": title,
+		"Data":  data,
+	})
 	if err != nil {
 		return err
 	}
 
-	f, _, err := openFile("dist/index.html")
+	http.ServeContent(w, r, "index.html", time.Now(), bytes.NewReader(bs))
+	return nil
+}
+
+var startTime = time.Now()
+
+func renderTemplate(name string, data any) ([]byte, error) {
+	f, _, err := openFile(name)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	bs, err := io.ReadAll(f)
 	_ = f.Close()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	bs = bytes.Replace(bs,
-		[]byte("window.POMERIUM_DATA = {}"),
-		append([]byte("window.POMERIUM_DATA = "), jsonData...),
-		1)
+	tpl, err := template.New("").Parse(string(bs))
```

Comment:
(optional)

Would it make sense to parse the template once (during startup), rather than for each call to `ServePage()`?